hibernate-core/src/main/java/org/hibernate/dialect/JsonHelper.java

```diff
@@ -1104,22 +1101,21 @@ private static Object fromString(
 			default:
 				if ( jdbcMapping.getJdbcType() instanceof AggregateJdbcType ) {
 					final AggregateJdbcType aggregateJdbcType = (AggregateJdbcType) jdbcMapping.getJdbcType();
-					final Object[] subValues = aggregateJdbcType.extractJdbcValues(
-							CharSequenceHelper.subSequence(
-									string,
-									start,
-									end
+					final StructAttributeValues subValues = StructHelper.getAttributeValues(
```

Comment:
This change looks wrong to me i.e. the `getAttributeValues` call should be moved into the `if ( returnEmbeddable )` branch.

Reply:
Done.